.github/workflows/js-test-and-release.yml

```diff
@@ -0,0 +1,153 @@
+name: test & maybe release
+on:
+  push:
+    branches:
+      - master # with #262 - ${{{ github.default_branch }}}
+  pull_request:
+    branches:
+      - master # with #262 - ${{{ github.default_branch }}}
+
+jobs:
+
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npm run --if-present lint
+      - run: npm run --if-present dep-check
+
+  test-node:
+    needs: check
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: [16]
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npm install -g node-pre-gyp
```

Comment:
I've added installation of `node-pre-gyp` here. Let me know if I should move it elsewhere.

Reply:
It looks like this needs to run after `actions/setup-node@v2`

See https://github.com/libp2p/js-libp2p-webrtc-direct/runs/5537246479?check_suite_focus=true

Don't now why PR CI did not fail though